### ISSUE TITLE
MessagingTracing OnCreateMessage() causes client initialization to fail when using EventListeners/ETW

### ISSUE THREAD

**Issue by rrector** [dotnet/orleans#6605]
I'm in the process of upgrading from 3.0.2 to 3.1.7.
Once upgraded to 3.1.7, calling ClusterClient.Connect() will always fail with the following exception:

```
System.ArgumentNullException: Value cannot be null.
Parameter name: grain
   at Orleans.Runtime.ActivationAddress.GetAddress(SiloAddress silo, GrainId grain, ActivationId activation)
   at Orleans.Runtime.Message.get_TargetAddress()
   at System.Diagnostics.DiagnosticSourceEventSource.TransformSpec.PropertySpec.PropertyFetch.RefTypedFetchProperty`2.Fetch(Object obj)
   at System.Diagnostics.DiagnosticSourceEventSource.TransformSpec.Morph(Object obj)
   at System.Diagnostics.DiagnosticSourceEventSource.FilterAndTransform.Morph(Object args)
   at System.Diagnostics.DiagnosticSourceEventSource.FilterAndTransform.<>c__DisplayClass2_1.<.ctor>b__2(KeyValuePair`2 evnt)
   at System.Diagnostics.DiagnosticSourceEventSource.CallbackObserver`1.OnNext(T value)
   at System.Diagnostics.DiagnosticListener.Write(String name, Object value)
   at Orleans.Runtime.MessagingTrace.OnCreateMessage(Message message)
   at Orleans.Runtime.MessageFactory.CreateMessage(InvokeMethodRequest request, InvokeMethodOptions options)
   at Orleans.OutsideRuntimeClient.SendRequest(GrainReference target, InvokeMethodRequest request, TaskCompletionSource`1 context, String debugContext, InvokeMethodOptions options, String genericArguments)
   at Orleans.Runtime.GrainReferenceRuntime.SendRequest(GrainReference reference, InvokeMethodRequest request, String debugContext, InvokeMethodOptions options)
   at Orleans.Runtime.OutgoingCallInvoker.<Invoke>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.Halo.Orleans.GrainCallFilters.LogOutgoingRequestGrainCallFilter.<Invoke>d__0.MoveNext() in D:\Repos\343.Services\Shared\Microsoft.Halo.Orleans\GrainCallFilters\LogOutgoingRequestGrainCallFilter.cs:line 38
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Halo.Orleans.GrainCallFilters.LogOutgoingRequestGrainCallFilter.<Invoke>d__0.MoveNext() in D:\Repos\343.Services\Shared\Microsoft.Halo.Orleans\GrainCallFilters\LogOutgoingRequestGrainCallFilter.cs:line 63
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.Runtime.OutgoingCallInvoker.<Invoke>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.Runtime.GrainReferenceRuntime.<InvokeWithFilters>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.OutsideRuntimeClient.<>c__DisplayClass56_0.<<StartInternal>b__1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Orleans.OutsideRuntimeClient.<<StartInternal>g__ExecuteWithRetries|56_2>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.OutsideRuntimeClient.<StartInternal>d__56.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.OutsideRuntimeClient.<Start>d__55.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.ClusterClient.<Connect>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.Halo.Orleans.Hosting.HaloGrainClient.<CreateClient>d__12.MoveNext() in D:\Repos\343.Services\Shared\Microsoft.Halo.Orleans\Hosting\HaloGrainClient.cs:line 122
```

It appears that when `OnCreateMessage` is called, a message is passed that doesn't have a `TargetAddress` defined. When [DiagnosticSourceEventSource tries to Morph the message](https://github.com/dotnet/runtime/blob/a755e742cad93fb173b47a95f68168c29ae95fb6/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs#L644), it reflects over `Message.TargetAddress`, which calls `ActivationAddress.GetAddress()` with a null target grain, which throws.

I am able to work around this by excluding the `Microsoft-Diagnostics-DiagnosticSource` provider from our ETW captures and EventListener logging.

**Comment by sergeybykov** [comment]
Is this fixed with #6626?

**Comment by rrector** [comment]
No  - This is caused by some new DiagnosticSource/DiagnosticListener code added in the same commit as the EventSource changes that caused #6617 (#6226<https://github.com/dotnet/orleans/pull/6226/files>), but I don’t really know what the intent for the DiagnosticListener code is, so I didn’t want to hazard a guess. I felt comfortable with making the EventSource changes that were in #6626.
